directory_api_client/testapiclient.py

```
@@ -0,0 +1,15 @@
+from directory_api_client.base import BaseAPIClient
+
+
+class DirectoryTestAPIClient(BaseAPIClient):
+
+    endpoints = {
+        'company_by_ch_id': 'testapi/company/{ch_id}/'
+    }
+
+    def get_company_by_ch_id(self, ch_id: str):
+        result = None
+        if ch_id:
```

the name `get_company_by_ch_id` implies `ch_id` will always be present. Why allow `ch_id` to not be present here?

---

This check covers a case when you pass `ch_id=None`. It's a typical boundary value test https://en.wikipedia.org/wiki/Boundary_testing

---

ah I see. If we wanted to solve that problem it seems better to check the expected condition is met, and raising an error if not.

Returning `None` if the precondition is not met really is hiding the error. The error should be shouted loudly so we can fix it. Also, making this function sometimes return a `Response`, and sometimes return `None` is inconsistent behaviour. The fewer degrees of freedom the less we have to think :)

we could do this:

```
def get_company_by_ch_id(self, ch_id: str):
    assert ch_id
    ...
```

but still, I wonder if this is overkill. What is `ch_id` is dictionary or a list, or a boolean `True`? should we also be checking for that? No, I think not :)

I would remove the check that `ch_id` is `None`.